File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if (argc != 2 || argv[1] < 0)
    {
        printf("GO AWAY.\n");
        return 1;
    }
    string p = get_string("plaintext: ");
    int k = atoi(argv[1]) % 26;
    int c[strlen(p)];
    int i = 0;
    printf("ciphertext: ");
    for (i = 0; i < strlen(p); i++)
    {
       c[i] = (p[i] + k);
        if ((c[i] < 91 && c[i] > 64) || (c[i]> 96 && c[i]<123))
        {
            c[i] = c[i] ;
              printf("%c", c[i]);
        }
        }
    printf("\n");
}

File: caesar.py
import sys
from cs50 import get_int, get_string
#from ctype import isalpha, isupper, islower

if len(sys.argv) != 2:
    print("Usage: python caesar.py k")
    False


key = int(sys.argv[1])


text = get_string("text: ")


for i in range(len(text)):
    if text[i].isalpha():
        x = ord(text[i])

        if text[i].isupper():
            x = x - 65
            y = (x + key) % 26
            y = y + 65


        if text[i].islower():
            x = x - 97
            y = (x + key) % 26
            y = y + 97
    z = chr(y)
    print(f"ciphertext: {z}", end="")
